src/Compiler/Utilities/lib.fs

```diff
@@ -589,14 +589,17 @@ type DisposablesTracker() =
 module ArrayParallel =
 
     let inline iteri f (arr: 'T []) =
-        let parallelOptions = ParallelOptions(MaxDegreeOfParallelism = max (min Environment.ProcessorCount arr.Length) 1)
-        try
-            Parallel.For(0, arr.Length, parallelOptions, fun i ->
-                f i arr[i]
-            ) |> ignore
-        with
-        | :? AggregateException as ex when ex.InnerExceptions.Count = 1 ->
-            raise(ex.InnerExceptions[0])
+        match arr with
+        | [||] -> ()
+        | arr ->
+            let parallelOptions = ParallelOptions(MaxDegreeOfParallelism = Environment.ProcessorCount)
```

Comment:
`max (min Environment.ProcessorCount arr.Length) 1` isn't needed after all? I guess `Parallel.For` handles it internally?

Reply:
max(x, 1) isn't needed because we make sure the number of items is >0, so unless processorcount=0, we're ok.

min(processorcount, itemCount) isn't needed, because it should never use more workers than work items.

So just processorCount should be enough.